Rails 4 compatibilty

vincent-the opened the pull request:
Hello,

@Azdaroth started a Rails-4 branch, to make SimpleCaptcha compatible with Rails 4 final.
We made some minor adjustments and are already running SimpleCaptcha in a Rails 4 production app now.

Maybe this is useful for other people as well.

Azdaroth committed:
remove support for set_table_name, attr_accesible, dynamic finders, changed version

Azdaroth committed:
fix including rails helpers

Azdaroth committed:
beack to previous version with rails sprockets

Azdaroth committed:
confusing change with rails 4 sprockets

Azdaroth committed:
lucky try with sprockets

Commit:
fixed ActiveRecord Naming reference for Rails 4.0.0rc2

av-ast commented:
+1

pmuens commented:
+1

freemanoid commented:
It would be great to accept this PR.

maxmumford commented:
+1

zealot128 committed:
merge conflict Sprockets

zealot128 committed:
merge conflict: set table name

zealot128 committed:
Dummyapp and integration tests added; SimpleCaptcha.always_pass = Rails.env.test? extracted

zealot128 committed:
additional test with model based methods

zealot128 committed:
travis.ci first try

zealot128 committed:
ruby 1.9.3

zealot128 committed:
Also added pg and mysql to test spec

zealot128 committed:
mysql2 instead of mysql

zealot128 committed:
Dummy app not static anymore: rake task for dynamic generation of dummy app

zealot128 committed:
removed dummy app from VCS

zealot128 committed:
finalized dummy rake task and travis build command

zealot128 committed:
Formtastic version 2.2 and Rails 3.2 tested and supported with new custom input.

zealot128 committed:
attr_protected only for Rails 3

zealot128 committed:
mysql support

freemanoid committed:
Add refresh button

zealot128 commented:
Everyone who is still interested: We fixed some issues, made some Travis CI specs, bumped the version number over on https://github.com/pludoni/simple-captcha. Unfortunatly, both galethub and the original author did not respond to my messages :(.

To use it, just reference the git:

``` ruby
gem 'simple_captcha', :git => 'git://github.com/pludoni/simple-captcha.git'
```

Contributions (preferable with tests) are very welcome :)

freemanoid committed:
Correct contributing part of README according to travis config

As I can see we _always_ need to use dummy app for testing.

freemanoid committed:
Add boot.rb with custom Gemfile path

zealot128 committed:
Merge pull request #4 from freemanoid/patch-1

Correct contributing part of README according to travis config

zealot128 committed:
Merge pull request #5 from freemanoid/default-gemfile-for-dummy

Fix default gemfile for dummy application

zealot128 committed:
renamed gem to pludoni-simple_captcha for now

zealot128 committed:
renamed to simple_captcha2

zealot128 committed:
drop Ruby 1.8.7

zealot128 committed:
converted README to markdown with syntax highlighting

zealot128 committed:
deleted old README, fixed markdown formatting issues

zealot128 committed:
...

maxmumford commented:
@zealot128 Just deployed your branch in production on rails 4 and it works great. Haven't thoroughly tested configuration combinations, but all cases in my basic scenario seem to be good. Thanks! It's about time we have a version 4 compatible captcha solution :)

pmuens commented:
Great work! Thank you very much!
Is it possible to get this via RubyGems because we have some requirements that we're not allowed to use Git repositories in the Gemfile for the Gems we want to use...

zealot128 commented:
Yes. I released simple_captcha2 with all the fixes and plan further 
development:
http://rubygems.org/gems/simple_captcha2

Am 03.12.2013 08:29, schrieb Philipp Müns:

> Great work! Thank you very much!
> Is it possible to get this via RubyGems because we have some 
> requirements that we're not allowed to use Git repositories in the 
> Gemfile for the Gems we want to use...
> 
> —
> Reply to this email directly or view it on GitHub 
> https://github.com/galetahub/simple-captcha/pull/39#issuecomment-29688700.

pmuens commented:
Awesome! Thank you :thumbsup:

zealot128 committed:
clarification: needs to require simple_captcha for now

zealot128 committed:
test usage

zealot128 committed:
Update README.md

zealot128 committed:
gemspec warning fixed; travis.yml now with ruby 2.1 and rails 4.1beta

zealot128 committed:
tests now work

zealot128 committed:
v.0.2.1: fixed bundle warning, missing README.rdoc

zealot128 committed:
maybe fix: uninitialized constant

soylent committed:
fix mass-assignment error when using protected_attributes

fmbiete committed:
Support for using the same captcha in multiple forms of the same page

Commit:
fix: avoid 'create_migration' method conflict

zealot128 committed:
version 0.3.0 -> Rails 4.1 compatibility fixes

zealot128 committed:
Travis.yml -> Ruby 2.1 and rails 4.1 preferred

zealot128 committed:
Ruby version 2.1.1

zealot128 committed:
simple_captcha_valid? idempotent - saves last result; fixes #13

zealot128 committed:
Merge branch 'master' of github.com:fmbiete/simple-captcha into fmbiete-master

zealot128 committed:
Merge branch 'fmbiete-master'

zealot128 committed:
merge

zealot128 committed:
Refresh Button: Adjustments with generic id, at least for image, default text

zealot128 committed:
formtastic

zealot128 committed:
travis test fix?

rubyconvict committed:
Fixes #14 - undefined method 'apply_simple_captcha' (Mongoid)

zealot128 committed:
Merge pull request #15 from rubyconvict/master

Fixes #14 - undefined method 'apply_simple_captcha' (Mongoid)

zealot128 committed:
version bump

rubyconvict committed:
Updated README.md

zealot128 committed:
Merge pull request #16 from rubyconvict/master

Updated README.md

wpp commented:
Thank you @zealot128 You are awesome!

mrhead committed:
Add possibility to set `implode` valude for the generated image

zealot128 committed:
Merge pull request #20 from mrhead/mrhead-add-implode-option

Add possibility to set `implode` valude for the generated image

Feuda committed:
add i18n example for refresh button text

zealot128 committed:
Merge pull request #23 from Feuda/patch-1

add i18n example for refresh button text

lionello committed:
Using absolute URL for img src to support inclusion in non-rails pages

lionello committed:
Remove the need for tempfiles

lionello committed:
Add noise to image

Conflicts:
	lib/simple_captcha/image.rb

zealot128 committed:
Merge pull request #28 from lionello/noise

Add noise to image

zealot128 committed:
Merge pull request #26 from lionello/absolute_url

Using absolute URL for img src to support inclusion in non-rails pages

zealot128 committed:
Merge pull request #27 from lionello/nofiles

Remove the need for tempfiles

incubus committed:
Remove the executable bit from image.rb

zealot128 committed:
Merge pull request #31 from incubus/master

Remove the executable bit from image.rb

zealot128 committed:
made refresh button work - request is not defined

fmbiete committed:
Fix warning
simple_captcha2-0.3.2/lib/simple_captcha/view.rb:136
warning: circular argument reference - request

zealot128 committed:
Merge pull request #35 from fmbiete/master

Ruby 2.2.0: Fix warning

incubus committed:
add additional check for request

zealot128 committed:
Merge pull request #36 from incubus/master

Add additional check for request

kuldeepaggarwal committed:
allow user to add customised response headers
customised partial_parth for captcha

zealot128 committed:
0.3.3

kuldeepaggarwal committed:
fetching time to microseconds

kuldeepaggarwal committed:
assign unique id to hidden captcha_key

kuldeepaggarwal committed:
extract simple_captcha options in another helper

kuldeepaggarwal committed:
extract url creation logic

zealot128 committed:
Merge pull request #39 from kuldeepaggarwal/cross-origin

Little improvements

zealot128 committed:
version 0.3.4

aliaksandrb committed:
Add missing default application.css to make tests pass

aliaksandrb committed:
Add optional possibility to generate captcha partial with HAML

aliaksandrb committed:
Add missed newlines at the end of files

zealot128 committed:
Merge pull request #40 from aliaksandrb/master

Add optional possibility to generate captcha partial with HAML

rubyconvict committed:
Remove obsolete info from README.md

zealot128 committed:
Merge pull request #41 from rubyconvict/master

Remove obsolete info from README.md

beviz committed:
fix no session request(api etc) will never delete expired captcha data

zealot128 committed:
Merge pull request #46 from beviz/master

fix no session request(api etc) will never delete expired captcha data

miguelnery commented:
Thank you very much : D  +1

Friend-LGA committed:
Add autocorrect = 'off' and autocapitalize = 'off' for iPhone/iPad

Friend-LGA committed:
Add autocorrect = 'off' and autocapitalize = 'off' for iPhone/iPad

zealot128 committed:
Travis: migrate to docker

zealot128 committed:
Added Changelog

zealot128 committed:
Added Rubygems Badge

zealot128 committed:
Merge branch 'master' of https://github.com/Friend-LGA/simple-captcha into Friend-LGA-master

zealot128 committed:
matched project style for hash keys

zealot128 committed:
Merge branch 'Friend-LGA-master'

zealot128 committed:
changed noise parameter: only added if noise is set and > 0

zealot128 committed:
Merge branch 'master' of https://github.com/soylent/simple-captcha into soylent-master

zealot128 committed:
Merge branch 'soylent-master'

zealot128 committed:
version bump to 0.4.0

zealot128 committed:
Updated Changelog

Commit:
set the format of the refresh procedure with javascript

Commit:
Test against latest Ruby

Commit:
Test against latest MRI patch levels

Commit:
Test against Rails 5 by default

Commit:
Fill in missing RDMBs

Commit:
Fix Rails 5 warnings

Commit:
Remove old rubies from Travis matrix

Commit:
Fix contributing instructions

Commit:
Rails 5 by default

Commit:
Add missing rails 5 dependency

Commit:
Fix Rails 5 tests

Commit:
Fix ruby warning

```
warning: loading in progress, circular require considered harmful
```

Commit:
Fix another ruby warning

```
File.exists? is a deprecated name, use File.exist? instead
```

Commit:
Fix more Rails deprecations in tests

```
DEPRECATION WARNING: `render :text` is deprecated because it does not
actually render a `text/plain` response. Switch to `render plain: 'plain
text'` to render as `text/plain`, `render html: '<strong>HTML</strong>'`
to render as `text/html`, or `render body: 'raw'` to match the
deprecated behavior and render with the default Content-Type, which is
`text/plain`.
```

Commit:
Add missing pg gem

Commit:
Allow ruby-head failures

Commit:
Don't mess with old formtastic versions

Test against the latest.

Commit:
Test the full matrix

Commit:
Fix state leak

Commit:
Comment out the rest of the skipped test

To prevent some "unused variable" warnings.

Commit:
Fix another unused variable warning

Commit:
Fix another warning

```
warning: instance variable @_simple_captcha_result not initialized
```

zealot128 committed:
Merge pull request #53 from deivid-rodriguez/rails_5_support

Rails 5 support

Commit:
Merge branch 'master' of git://github.com/pludoni/simple-captcha

bfbones committed:
use symbols instead of strings for configuring refresh_format, define available refresh_formats and raise error if adapter not available, cleaned up code

bfbones committed:
bumped version

zealot128 committed:
Added new JavaScript Adapter plain_javascript

zealot128 committed:
Changelog with 0.4.2 changes

denschub committed:
Allow manual font family overrides
would allow people to fix the issues raised in #52

zealot128 committed:
Merge pull request #55 from denschub/im697

Allow manual font family overrides

zealot128 committed:
Version 0.4.3 Bump and Changelog

denschub committed:
Specify the font before the label to avoid issues

Commit:
Fix headers in Readme.md

philippneugebauer committed:
added debian hint

zealot128 committed:
Merge pull request #61 from philippneugebauer/master

added debian hint

Commit:
Support Sequel ORM

zealot128 committed:
Merge branch 'master' into patch-1

zealot128 committed:
Merge pull request #60 from Snuff/patch-1

Fix headers in Readme.md

zealot128 committed:
Merge pull request #57 from denschub/reorder-font-config

Specify the font before the label to avoid issues

Commit:
Support running tests with Rails 5.2

zealot128 committed:
Merge pull request #64 from cmrd-senya/rails-5.2

Support running tests with Rails 5.2

Commit:
Support testing Sequel ORM

Commit:
Document using different ORMs with the gem

zealot128 committed:
Merge pull request #63 from cmrd-senya/support-sequel

Support Sequel ORM

dapi committed:
DeadLock in captcha deletion dosn't break process

zealot128 committed:
Merge pull request #65 from BrandyMint/master

DeadLock in captcha deletion dosn't break process

zealot128 committed:
Cut new release; Added installation notes about font

Nghi93 committed:
add license

zealot128 committed:
Merge pull request #68 from Nghi93/feature/add_license

add license
